src/app/view/perspectives/post-it/post-it-perspective.component.ts

```diff
@@ -17,465 +17,271 @@
+import {selectDocumentsByCustomQuery} from '../../../core/store/documents/documents.state';
+import {QueryModel} from '../../../core/store/navigation/query.model';
 import {PostItLayout} from '../../../shared/utils/layout/post-it-layout';
 import {PostItLayoutConfig} from '../../../shared/utils/layout/post-it-layout-config';
-import {AttributePropertySelection} from './document-data/attribute-property-selection';
-import {Direction} from './document-data/direction';
-import {DocumentModel} from './document-data/document-model';
-import {PostItDocumentComponent} from './document/post-it-document.component';
+import {PostItSortingLayout} from '../../../shared/utils/layout/post-it-sorting-layout';
+import {InfiniteScroll} from './bussiness/infinite-scroll';
+import {NavigationManager} from './bussiness/navigation-manager';
+import {SelectionManager} from './bussiness/selection-manager';
+import {PostItDocumentModel} from './document-data/post-it-document-model';
+import Create = DocumentsAction.Create;
+import Delete = DocumentsAction.Delete;
+import UpdateData = DocumentsAction.UpdateData;
 
 @Component({
   selector: 'post-it-perspective',
   templateUrl: './post-it-perspective.component.html',
-  styleUrls: ['./post-it-perspective.component.scss'],
-  host: {
-    '(document:click)': 'onClick($event)'
-  }
+  styleUrls: ['./post-it-perspective.component.scss']
 })
 export class PostItPerspectiveComponent implements OnInit, OnDestroy {
 
   @Input()
   public editable: boolean = true;
 
-  @Input()
-  public useOwnScrollbar: boolean = false;
-
-  @ViewChild('layout')
-  public layoutElement: ElementRef;
-
-  @ViewChildren(PostItDocumentComponent)
-  public documentComponents: QueryList<PostItDocumentComponent> = new QueryList();
+  private _useOwnScrollbar = false;
 
-  public postIts: DocumentModel[] = [];
+  @Input()
+  public get useOwnScrollbar(): boolean {
+    return this._useOwnScrollbar;
+  }
 
-  public lastClickedPostIt: DocumentModel;
+  public set useOwnScrollbar(value: boolean) {
+    this._useOwnScrollbar = value;
 
-  public fetchingData: boolean;
+    if (this.infiniteScrollManager) {
+      this.infiniteScrollManager.setUseParentScrollbar(value);
+    }
+  }
 
-  public collections: { [collectionCode: string]: Collection } = {};
+  @ViewChild('layout')
+  public layoutElement: ElementRef;
 
-  public currentCollection: string;
+  public perspectiveId: string;
 
-  public query: Query;
+  public postIts: PostItDocumentModel[] = [];
 
-  private workspace: Workspace;
+  public infiniteScrollManager: InfiniteScroll;
 
-  private attributeSuggestions: { [collectionCode: string]: string[] } = {};
+  public navigationManager: NavigationManager;
 
-  private onInfiniteScroll: () => void | null;
+  public selectionManager: SelectionManager;
 
-  private appStateSubscription: Subscription;
+  private layoutManager: PostItLayout;
 
-  private layout: PostItLayout;
+  private pageSubscriptions: Subscription[] = [];
```

Comment:
Very nice concept of managing various subscriptions. But isn't there only a single one in this component?

Or does it mean that we store multiple instances of the same document subscription? We must not do that! We need to unsubscribe every time a new one is created. Otherwise, the performance of the application will drop significantly after working with it for a while.

Reply:
Well there are more subscriptions. 1 for each loaded page

So for example at start 1 subsciption to page 1 (documents 0 .. 2*rowSize) would be made. When user scolls down, action to fetch documents 2*rowSize + 1.. (2*rowSize * 1) + 2*rowSize and subscription on them would be added... when the user scroll down again, another page would be fetched and subscribed on.

Reply:
Nice one! I'm just thinking about the case when you change the query and stay in the same perspective. What will happen with these subscriptions then? How would the whole component behave in such a case?

It is not possible to do that now. However, in the (very near) future we would like to be able to change the query without redirecting to search perspective every time. And our components should be ready for that so no refactoring will be needed.

Reply:
Well in search perspective the refreshing works just fine, when changing queries.